File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    public int kthSmallest(TreeNode root, int k) {
        return kthSmallestRec(root, k)[1];
    }

    public int[] kthSmallestRec(TreeNode root, int k) {
        if (root == null) return new int[]{k, -1};

        var left = kthSmallestRec(root.left, k);
        if (left[0] == -1) return left;
        if (left[0] == 1) {
            return new int[]{-1, root.val};
        }

        var right = kthSmallestRec(root.right, left[0] - 1);
        if (right[0] == -1) return right;

        return new int[]{right[0], -1};
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        const result = [];
        if (!root) return result;

        const queue = [];

        queue.push(root);

        while (queue.length) {

            const n = queue.length;
            const levelResult = [];
            for (let i = 0; i < n; i++) {


                const node = queue.shift();
                levelResult.push(node.val);

                if (node.left) queue.push(node.left);
                if (node.right) queue.push(node.right);
            }
            result.push(levelResult);
        }

        return result;
    }
}
